my-page-app/src/components/admin/Admin.jsx

```diff
@@ -1,129 +1,165 @@
 import "./Admin.scss";
-import {useEffect, useState} from "react";
+import React, {useEffect, useState} from "react";
 import apiService from "../../services/api.service";
 import {toast} from "react-toastify";
 import {Spinner, Table} from "react-bootstrap";
+import Budgets from "../budget/Budgets";
 
 function Admin() {
-  const [users, setUsers] = useState([]);
-  const [budgetTypes, setBudgetTypes] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
+    const [users, setUsers] = useState([]);
+    const [budgetTypes, setBudgetTypes] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [expandedUser, setExpandedUser] = useState(null);
+    const [filterValue, setFilterValue] = useState("");
 
-  useEffect(() => {
-    refreshTable();
-    // eslint-disable-next-line
-  }, []);
+    useEffect(() => {
+        refreshTable();
+        // eslint-disable-next-line
+    }, []);
 
-  const extractListOfBudgets = (users) => {
-    if (users.length > 0){
-      let extractedBudgetTypes = [];
-      users.map(user => user.budgets.forEach(budget =>
-          {
-            if (!budgetTypeListContains(extractedBudgetTypes, budget)) {
-              budget.budgetType.balanceIsHours = false;
-              extractedBudgetTypes.push(budget.budgetType);
-              if (budget.budgetType.allowTimeBalance) {
-                let budgetTypeHours = Object.assign({}, budget.budgetType);
-                budgetTypeHours.balanceIsHours = true;
-                budgetTypeHours.name += "(timer)";
-                extractedBudgetTypes.push(budgetTypeHours);
-              }
-            }
+    const extractListOfBudgets = (users) => {
+        if (users.length > 0) {
+            let extractedBudgetTypes = [];
+            users.map(user => user.budgets.forEach(budget => {
+                    if (!budgetTypeListContains(extractedBudgetTypes, budget)) {
+                        budget.budgetType.balanceIsHours = false;
+                        extractedBudgetTypes.push(budget.budgetType);
+                        if (budget.budgetType.allowTimeBalance) {
+                            let budgetTypeHours = Object.assign({}, budget.budgetType);
```

Comment:
Foretrekk alltid `const` framfor `let` med mindre man har primitive verdier eller objekter som kan re-assignes helt.
```suggestion
            const extractedBudgetTypes = [];
            users.map(user => user.budgets.forEach(budget => {
                    if (!budgetTypeListContains(extractedBudgetTypes, budget)) {
                        budget.budgetType.balanceIsHours = false;
                        extractedBudgetTypes.push(budget.budgetType);
                        if (budget.budgetType.allowTimeBalance) {
                            const budgetTypeHours = Object.assign({}, budget.budgetType);
```

De siste 2 årene har jeg skrevet `let` maks 10 ganger 😃